library/java/test/io/envoyproxy/envoymobile/engine/JvmBridgeUtilityTest.kt

```diff
@@ -0,0 +1,58 @@
+package io.envoyproxy.envoymobile.engine
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class JvmBridgeUtilityTest {
+
+  @Test
+  fun `retrieveHeaders produces a Map with all headers provided via passHeaders`() {
+    val utility = JvmBridgeUtility()
+    utility.passHeader("test-0".toByteArray(), "value-0".toByteArray(), true)
+    utility.passHeader("test-1".toByteArray(), "value-1".toByteArray(), false)
+    utility.passHeader("test-1".toByteArray(), "value-2".toByteArray(), false)
+
+    val headers = utility.retrieveHeaders()
+    val expectedHeaders = mapOf(
+      "test-0" to listOf("value-0"),
+      "test-1" to listOf("value-1", "value-2")
+    )
+
+    assertThat(headers)
+      .hasSize(2) // Two keys / header name
+      .usingRecursiveComparison().isEqualTo(expectedHeaders)
```

Comment:
```suggestion
      .usingRecursiveComparison()
      .isEqualTo(expectedHeaders)
```

Reply:
These two are actually dependently-chained (and you couldn't, for instance, reverse the order of 22 and 23, since 23 changes the type) which was why I put them on one line. I don't feel terribly strongly about it though.